From 0bdf3119f1610a6cdc3f53545810d965eb0490ea Mon Sep 17 00:00:00 2001
From: "sameer.mulla" <sameer.mulla@lge.com>
Date: Mon, 1 Oct 2018 10:49:38 +0530
Subject: [PATCH] Fix obexd segmentation fault

:Release Notes:
Fix obexd segmentation fault

:Detailed Notes:
cancel ongoing transfer from obexctl command line by below commands
obexctl> connect
obexctl> send /tmp/sample.mp3
obexctl> cancel /org/bluez/obex/client/session0/transfer0
segmentation fault observed in obexd
Backtrace
err=0x5555557ee900) at gobex/gobex-transfer.c:103
err=<optimized out>, rsp=0x0, user_data=0x5555557f6ef0) at
gobex/gobex-transfer.c:221
gobex/gobex.c:262
/lib/x86_64-linux-gnu/libglib-2.0.so.0
/lib/x86_64-linux-gnu/libglib-2.0.so.0
/lib/x86_64-linux-gnu/libglib-2.0.so.0
/lib/x86_64-linux-gnu/libglib-2.0.so.0
out>) at obexd/src/main.c:322

:Testing Performed:
Tested all obexctl commands

:QA Notes:

:Issues Addressed:
[PLAT-66851] Fix segmentation fault in bluez while cancelling
             of opp transfer

Change-Id: Ic3fbb37fc52cb84ee32286dc5edb88a20eba3d67

---
 gobex/gobex-transfer.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/gobex/gobex-transfer.c b/gobex/gobex-transfer.c
index bc99306..36d4a1c 100644
--- a/gobex/gobex-transfer.c
+++ b/gobex/gobex-transfer.c
@@ -100,10 +100,11 @@ static void transfer_complete(struct transfer *transfer, GError *err)
 
 	g_obex_debug(G_OBEX_DEBUG_TRANSFER, "transfer %u", id);
 
+	        /* Check if the complete_func removed the transfer */
+        if (find_transfer(id) == NULL)
+                return;
+
 	transfer->complete_func(transfer->obex, err, transfer->user_data);
-	/* Check if the complete_func removed the transfer */
-	if (find_transfer(id) == NULL)
-		return;
 
 	transfer_free(transfer);
 }
